numpy/lib/_function_base_impl.py

```diff
@@ -4880,6 +4883,14 @@ def trapz(y, x=None, dx=1.0, axis=-1):
     >>> np.trapz(a, axis=1)
     array([2.,  8.])
     """
+
+    # Deprecated in NumPy 2.0, 2023-08-18
+    warnings.warn(
```

Comment:
FWIW, we did that same rename in `scipy.integrate` a few years ago, and got no complaints. This name is considered a slur, and that's an extra reason to get rid of it. We touched on this in the 2.0 dev call IIRC, and folks were in favor of names that are both more descriptive and more inclusive (`cumsum` & co too).

It's deprecated rather than outright removed here exactly because it's used quite a bit. So no code will actually break in the 2.0 release.

Reply:
Huh, didn't know it's a slur. Thanks for the clarification!

Reply:
Neither did I under https://github.com/scipy/scipy/issues/12924.